src/PAModel/Checksum/ChecksumMaker.cs

```diff
@@ -189,7 +189,8 @@ internal static string ChecksumToString(byte[] bytes)
 
             foreach (var kv in _files.OrderBy(x => x.Key))
             {
-                hash.AppendData(kv.Value);
+                var fileNameInBytes = Encoding.ASCII.GetBytes(kv.Key);
+                hash.AppendData(fileNameInBytes.Concat(kv.Value).ToArray());
```

Comment:
Why this change?
This looks like a change to the checksum algorithm, which can be significant (server computes checksum too).

Reply:
we were calculating whole checksum based on the content of the files, not account for the different filenames due to which the issue that my changes caused slipped through. I made this change to calculate the checksum for content of the file along with filename to ensure everything match.

Reply:
I see - we're not including kv.Key in the hash. 

can we just add:
`hash.AppendData(kv.Key);`

and avoid the concat/toArray?